examples/machine_translation/inference.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import tensorflow as tf
+from absl import app
+from absl import flags
+from absl import logging
+
+import examples.machine_translation.data  # noqa: F401
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer(
+    "sequence_length",
+    20,
+    "Input and output sequence length.",
+)
+
+flags.DEFINE_string(
+    "saved_model_path",
+    "saved_models/machine_translation_model",
+    "The path to saved model",
+)
+
+flags.DEFINE_string("inputs", None, "The inputs to run machine translation on.")
+
+EXAMPLES = [
+    (
+        "Tom doesn't listen to anyone.",
+        "[start] Tomás no escucha a nadie. [end]",
+    ),
+    ("I got soaked to the skin.", "[start] Estoy chorreando. [end]"),
+    ("I imagined that.", "[start] Me imaginé eso. [end]"),
+    ("The baby is crying.", "[start] El bebé está llorando. [end]"),
+    (
+        "I've never felt so exhilarated.",
+        "[start] Nunca me he sentido tan animado. [end]",
+    ),
+    (
+        "Please forgive me for not having written sooner.",
+        "[start] Perdóname por no haberte escrito antes, por favor. [end]",
+    ),
+    ("I expected more from you.", "[start] Esperaba más de vos. [end]"),
+    ("I have a computer.", "[start] Tengo un computador. [end]"),
+    ("Dinner's ready!", "[start] ¡La cena está lista! [end]"),
+    ("Let me finish.", "[start] Déjame terminar. [end]"),
+]
+
+
+def decode_sequence(input_sentence, model, max_sequence_length, lookup_table):
```

Comment:
We could consider reworking this to actually use WordPiece tokenization. That would remove the need for a lot of complexity here (registering serializables, this entire function, rebuilding an inverse lookup table). You would need to download a vocabulary rather then adapt (could use the bert vocabulary I use for that example).

Might be overall more readable. Not saying we have to though, just something to consider.

Reply:
This is a good point, I will create an issue and do it in a separate PR. 

One question - does wordpiece tokenization support this autoregressive text generation?

Reply:
I think the usually approach would be to do the "autoregressive" part in token id space (keep a running list of token ids). And then detokenize a single sample/paragraph/etc at a time.

You could detokenize one token at a time if you wanted, but it would be less efficient, and you would probably need to check for the suffix indicator in each piece yourself.

Reply:
Follow up sounds good to me!